packages/app/src/DashboardFiltersEditModal.tsx

```diff
@@ -0,0 +1,284 @@
+                <Radio.Group
+                  value={value}
+                  onChange={v => onChange(v)}
+                  withAsterisk
+                >
+                  <Group>
+                    {metricTypes.map(type => (
+                      <Radio key={type} value={type} label={type} />
+                    ))}
+                  </Group>
+                </Radio.Group>
+              )}
+            />
+          </Input.Wrapper>
+        )}
+
+        <Input.Wrapper
+          label="Filter Expression"
+          description="SQL column or expression to filter on"
+          required
+        >
+          <SQLInlineEditorControlled
+            tableConnections={tableConnection}
+            control={control}
+            name="expression"
+            placeholder="SQL column or expression"
+            language="sql"
+            enableHotkey
+            rules={{ required: true }}
+            parentRef={parentRef}
+          />
+        </Input.Wrapper>
+
+        <Group justify="space-between" mt="md">
+          <Button
+            variant="outline"
+            color="red"
+            onClick={() => onRemoveFilter(filter.id)}
+          >
+            Delete
+          </Button>
+          <Button type="submit" className="align-self-end">
+            Save
+          </Button>
+        </Group>
+      </Stack>
+    </form>
+  );
+};
+
+interface EmptyStateProps {
+  onCreateFilter: () => void;
+}
+
+const EmptyState = ({ onCreateFilter }: EmptyStateProps) => {
+  return (
+    <Stack align="center" justify="center" py="xl">
+      <Text size="md" maw={300} ta="center">
+        Dashboard filters allow users of this dashboard to quickly filter on
+        important columns. Filters are saved in the dashboard.
+      </Text>
+      <Button variant="outline" onClick={onCreateFilter}>
+        Create Filter
+      </Button>
+    </Stack>
+  );
+};
+interface DashboardFiltersEditModalProps {
+  opened: boolean;
+  onClose: () => void;
+  filters: DashboardFilter[];
+  onSaveFilter: (filter: DashboardFilter) => void;
+  onRemoveFilter: (id: string) => void;
+  parentRef?: HTMLElement | null;
+}
+
+const NEW_FILTER_ID = 'new';
+
+const DashboardFiltersEditModal = ({
+  opened,
+  onClose,
+  filters,
+  onSaveFilter,
+  onRemoveFilter,
+}: DashboardFiltersEditModalProps) => {
+  const [selectedFilter, setSelectedFilter] = useState<
+    DashboardFilter | undefined
+  >(filters[0]);
+
+  useEffect(() => {
+    if (opened) {
+      setSelectedFilter(filters[0]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [opened]);
+
+  const handleRemoveFilter = (id: string) => {
+    if (id === selectedFilter?.id) {
+      setSelectedFilter(filters.find(f => f.id !== id));
+    }
+    onRemoveFilter(id);
+  };
+
+  const handleAddNewFilter = () => {
+    setSelectedFilter({
+      id: NEW_FILTER_ID,
+      type: 'QUERY_EXPRESSION',
+      name: 'New Filter',
```

Comment:
UX Wise, I think this should be an empty string and a placeholder called `Filter Name` or something should be set, it seems weird to have to copy and delete the name when typing. If this was intentional, thats fine though

Reply:
Nope, that wasn't intentional - replaced with an empty string on the object and a placehold in the input